GNUmakefile

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 The Cockroach Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+all: build
+	$(MAKE) help
+
+help:
+	@echo
+	@echo "Tip: use ./dev instead of 'make'."
+	@echo "Try:"
+	@echo "    ./dev help"
+	@echo
+
+# Generic build rules.
+.PHONY: build build%
+build: doctor
```

Comment:
Let's not have this direct dependency on `doctor` in this target or any others.

If you run `dev build` without running `doctor`, you'll get an error and it'll ask you to run `doctor`. Meanwhile, `dev build` does a (lightweight) check whether you've already run `doctor` every time you run it.

However, the way this dependency is structured means that we must unconditionally run `dev doctor` every time before building, which is rather slow.

Reply:
> Let's not have this direct dependency on doctor in this target or any others.

Fixed.